lupa/_lupa.pyx

```diff
@@ -1446,20 +1451,48 @@ cdef int py_object_gc(lua_State* L) nogil:
 
 cdef bint call_python(LuaRuntime runtime, lua_State *L, py_object* py_obj) except -1:
     cdef int i, nargs = lua.lua_gettop(L) - 1
+    cdef _LuaTable table
+    cdef bytes encoding
     cdef tuple args
+    cdef dict kwargs
 
     if not py_obj:
         raise TypeError("not a python object")
 
     f = <object>py_obj.obj
 
-    if not nargs:
+    if nargs == 0:
         lua.lua_settop(L, 0)  # FIXME
         result = f()
     else:
-        arg = py_from_lua(runtime, L, 2)
+        kwargs = {}
+        
+        if lua.lua_istable(L, -1) and lua.lua_getmetatable(L, -1):
```

Comment:
This part looks mostly like `_fix_args_kwargs()` spelled differently. I don't see a need for the duplication.

Reply:
This is way different than `_fix_args_kwargs`.

`_fix_args_kwargs` checks if the first argument is a table and unpacks it.
`call_python` checks if the last argument is a table with our hidden metatable and unpacks it.

Reply:
`_fix_args_kwargs` takes a Lua table and splits it into positional and keyword arguments for a Python call. `call_python` (after your change) takes a Lua table and splits it into positional and keyword arguments for a Python call. Where exactly is the difference?

Reply:
On `_fix_args_kwargs`, table entries accept integer keys as positional arguments, while on `call_python` (after the change), table entries must have string entries only.

Reply:
But we could create a `_LuaTable.to_dict` method which converts it to a dictionary, so that we could soak the repeated code from both functions, what do you think?

Reply:
sorry, right, in the case of `call_python`, the positional arguments come from the stack, not from a table. And as I said, it's better to prepare the call dict upfront.

Why not have a general function `python.pyargs(lua_table)` that returns an object that a Python call unpacks into an args/kwargs pair? Then you could just say
```lua
python.builtins.print(python.pyargs{1, 2, 3, sep=";"})
```
To me, that seems more versatile and more readable than requiring to split the arguments into "normal" Lua arguments and "special" Python keyword arguments _within the same call_.

Reply:
The only problem I find with this is that merging positional arguments seems inappropriate, unless we are willing to admit this is the only argument being passed.

Reply:
```python
>>> print(*(1,2,3), *(4,5,6))
1 2 3 4 5 6
```

Reply:
Fair enough.